indexer/drules_selector.cpp

```diff
@@ -153,6 +153,8 @@ bool GetBoundingBoxArea(FeatureType & ft, double & sqM)
   if (feature::GeomType::Area != ft.GetGeomType())
     return false;
 
+  // FIXME: the best geometry is loaded here always, use the current zoom level
+  // see https://github.com/organicmaps/organicmaps/issues/2840
   sqM = mercator::AreaOnEarth(ft.GetLimitRect(scales::GetUpperScale()));
```

Comment:
Hm, good catch. In fact, we killed the initial different zooms idea here.
I see that zoom is already passed into BaseRule::TestFeature. Can you restore it here?

Reply:
I looked at it awhile ago, so don't remember the details atm..